docs/computing/containers/overview.md

```diff
@@ -1,71 +1,251 @@
-
-- [Creating HPC-friendly containers](creating.md)
-- [Running existing containers](run-existing.md)
-- [Easy-to-use container wrapper tool](tykky.md)
+---
+description: Instructions for building and running Apptainer containers in CSC supercomputers.
+---
+
+# Apptainer containers
+
+In this section, we provide instructions on how to build and run containers using [Apptainer](https://apptainer.org/) with fakeroot enabled in HPC clusters without unprivileged user namespaces.
+We explain the special aspects of building and running containers on Puhti and Mahti clusters including how to set up the build environment, how to invoke the build commands, how to write container definition files and how to run containers.
+For general instructions about building and running containers, we recommend that users read the official [Apptainer documentation](https://apptainer.org/docs/user/main/index.html).
+
+Some reasons for using containers on HPC clusters are:
+
+- Containers improve startup times and avoid I/O bottlenecks with the parallel file system, that is Lustre in Puhti and Mahti, for applications that consist of a large number of files or load many shared libraries on startup such as Python, R, and MATLAB.
+  This is because Apptainer uses the Singularity Image Format (SIF) which packs the container files into a single SquashFS file.
+
+- Running containerized software is reproducible because the container image is immutable.
+
+- Container build definitions capture more comprehensively what is installed into the container and how it is installed.
+
+Limitations of containers:
+
+- Containers are not composable and they are not a substitute for package managers.
+  That is, having a container with Python installed and another container with R installed, will not provide you with a container with both Python and R installed.
+
+## Running containers
+
+### Using Apptainer directly
+
+Assume we have a container image called `container.sif`.
+We can execute an arbitrary command (replace `mycommand`) inside the container using the `apptainer exec` command as follows:
+
+```bash
+apptainer exec container.sif mycommand
+```
+
+We can make directories from the host available inside the container by using bind mounts.
+On Puhti and Mahti, we can bind mount the different [Disk Areas](../disk.md) to the container as follows:
+
+```bash
+apptainer exec --bind="/users,/projappl,/scratch,$TMPDIR,$LOCAL_SCRATCH" container.sif mycommand
+```
+
+We can add Nvidia GPU support with the `--nv` flag as follows:
+
+```bash
+apptainer exec --nv container.sif mycommand
+```
+
+We can use the same flags with `apptainer run` and `apptainer shell` commands.
+
+### Using Apptainer wrapper
+
+Many CSC provided software environments that use containers provide access via the `apptainer_wrapper` script.
+The wrapper uses environment variables to find the path to the container image (`SING_IMAGE`) and to provide flags (`SING_FLAGS`) such as `--nv`.
```

Comment:
Should there be a link somewhere here to all apptainer support env variables in apptainer docs?